### ISSUE TITLE
[PATCH] dirlist shows urlencoded uri in h1 tag

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10404]
|<img alt="Jerub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Jerub reported|
|-|-|
|Trac ID|trac#515|
|Type|defect|
|Created|2004-02-20 10:44:44Z|

Attachments:

* [patch-dirlist.htm](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f5b/f5b68d74621aa3616444d8ac07c9052a6030cf31/aacff9316cb898f4a481860faa395e751facf3d7.htm) (636 bytes) - added by Jerub on 2004-02-20 10:58:25Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__515 515
type__defect defect
reporter__Jerub Jerub
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1077273884000000 1077273884000000
changetime__1471776196530402 1471776196530402
version__ 
owner__ 
cc__Jerub
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Jerub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Jerub commented|
|-|-|

```
#!html
<pre>
Currently dirlist doesn't unquote the uri before displaying it in a h1 tag. The
uri potentially has strings like '%20' (space) and '%7E' (tilde).

This patch calls urllib.unquote on the string which is the title of the page.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)|@teratorn commented|
|-|-|

```
#!html
<pre>
committed. Thanks :)
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc commented|
|-|-|

In changeset ab35b6ce88019c4436bb2542f8b144a28eae89cf
```
#!CommitTicketReference repository="" revision="ab35b6ce88019c4436bb2542f8b144a28eae89cf"
Merge pull request #515 from twisted/8777-rodrigc-gzipencoder-py3

Author: rodrigc
Reviewer: charmander
Fixes: #8777

Enable GzipEncoderTests and AllowedMethodTests on Python 3
```
